### PR TITLE
#63 Add search improvments

### LINKED ISSUE #63: Search on the Search page
- [x] Send data to a backend

- [x] Fetch data from the database

- [x] Send data to a backend

- [x] Send data from backend to frontend

### PR DESCRIPTION
**Pull request by Pr1sm0**

### CODE REVIEW

File: frontend/src/components/pages/search/Search.tsx [hunk 1]
```diff
@@ -41,8 +43,16 @@ const Search: React.FC = () => {
...
   const getGender = () => {
-    if ((!checkboxFilters.female && !checkboxFilters.male) || (checkboxFilters.female && checkboxFilters.male)) {
+    //TODO: right organization for genders
+    if (
+      (!checkboxFilters.female && !checkboxFilters.male && !checkboxFilters.other) ||
+      (checkboxFilters.female && checkboxFilters.male && checkboxFilters.other)
+    ) {
       return;
     }
     return Boolean(checkboxFilters.male) ? 0 : 1;
```
Comment (oleksandrstarov) [inline frontend/src/components/pages/search/Search.tsx]: other is not considered here

File: frontend/src/components/pages/search/Search.tsx [hunk 2]
```diff
@@ -41,8 +43,16 @@ const Search: React.FC = () => {
     setSelectsFilters({ ...selectsFilters, [e.target.name]: e.target.value });
   };
 
+  useEffect(() => {
+    apiService(dataType.users).then(users => setUsers(users));
```
Comment (oleksandrstarov) [inline frontend/src/components/pages/search/Search.tsx]: please use await

File: frontend/src/components/pages/search/Search.tsx [hunk 3]
```diff
@@ -41,8 +43,16 @@ const Search: React.FC = () => {
...
+  }, []);
+
   const getGender = () => {
-    if ((!checkboxFilters.female && !checkboxFilters.male) || (checkboxFilters.female && checkboxFilters.male)) {
+    //TODO: right organization for genders
+    if (
+      (!checkboxFilters.female && !checkboxFilters.male && !checkboxFilters.other) ||
+      (checkboxFilters.female && checkboxFilters.male && checkboxFilters.other)
+    ) {
```
Comment (oleksandrstarov) [inline frontend/src/components/pages/search/Search.tsx]: const genderFilters = Object.values(checkboxFilters);

if (genderFilters.every(e=>e) || genderFilters.every(e=>!e)) { return; }

File: frontend/src/components/pages/search/Search.tsx [hunk 4]
```diff
@@ -71,22 +81,23 @@ const Search: React.FC = () => {
     apiService(dataType.filter, requestBody).then(users => setUsers(users));
   };
 
-  const years = [...Array(hightLimit - lowLimit)].map((_, index) => index + lowLimit);
+  const lowYears = [...Array(hightLimit - lowLimit)].map((_, index) => index + lowLimit);
+  const hightYears = [...Array(hightLimit - lowLimit)].map((_, index) => hightLimit - index);
```
Comment (oleksandrstarov) [inline frontend/src/components/pages/search/Search.tsx]: no need to do the same, maybe we can const hightYears = [... lowYears].reverse()?

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

**Comment by Pr1sm0** [comment]
Could you have a look now, please?

File: frontend/src/components/pages/search/Search.tsx [hunk 5]
```diff
@@ -33,19 +36,43 @@ const Search: React.FC = () => {
...
 
   const handleSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
     setSelectsFilters({ ...selectsFilters, [e.target.name]: e.target.value });
   };
 
-  const getGender = () => {
-    if ((!checkboxFilters.female && !checkboxFilters.male) || (checkboxFilters.female && checkboxFilters.male)) {
+  useEffect(() => {
+    async function getUsers() {
+      await apiService(dataType.users).then(users => setUsers(users));
```
Comment (oleksandrstarov) [inline frontend/src/components/pages/search/Search.tsx]: the point was to 

const user = await apiService(dataType.users);

setUsers(users)

File: frontend/src/components/pages/search/Search.tsx [hunk 6]
```diff
@@ -33,19 +36,43 @@ const Search: React.FC = () => {
...
-    return Boolean(checkboxFilters.male) ? 0 : 1;
+    if (genders.male) {
+      arrGenders.push(0);
+    }
+    if (genders.female) {
+      arrGenders.push(1);
+    }
+    if (genders.other) {
+      arrGenders.push(2);
+    }
```
Comment (oleksandrstarov) [inline frontend/src/components/pages/search/Search.tsx:75]: I think we need to use Enum here instead of plain numbers

**Review by oleksandrstarov — APPROVED** [review]
Please check the comments

**Pull request merged**